### PR TITLE
Demo Chapter 5

### PR DESCRIPTION
**Pull request by ohsnapitskenny**
See README for more information. This is the Demo of Chapter 5. If I got the time, I'll update this branch for the demo of chapter 6

### CODE REVIEW

File: 1733Sprint/1-Kenny/Demo.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import vehicles.*;
```
Comment (ssoegriemsingh) [inline 1733Sprint/1-Kenny/Demo.java]: It's common practice to not use a * for your imports: https://stackoverflow.com/questions/147454/why-is-using-a-wild-card-with-a-java-import-statement-bad

File: 1733Sprint/1-Kenny/Demo.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	private List<Vehicle> vehicles = new ArrayList<Vehicle>();
+    // Main
+    public static void main(String args[]) {
+        Demo demo = new Demo();
+    }
+
+    // Constructor
+    public Demo() {
+        greeting();
+    	createList();
```
Comment (ssoegriemsingh) [inline 1733Sprint/1-Kenny/Demo.java]: Needs a space?

File: 1733Sprint/1-Kenny/Demo.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        	);
+    }
+
+    private void showVehicles() {
+        for (Vehicle v : vehicles) {
+            System.out.println("----------------------------------------------------------------");
+            v.move();
+            
+            if (v instanceof Electric) {
+                System.out.println("Remaining battery: " + ((Electric)v).getBatteryLevel() + "%");
```
Comment (ssoegriemsingh) [inline 1733Sprint/1-Kenny/Demo.java]: You are downcasting here. Can this be avoided?

File: 1733Sprint/1-Kenny/vehicles/AstonMartin.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
+package vehicles;
+import vehicles.*;
+
+public class AstonMartin extends Car implements Fuel {
+  	double fuelLevel;
+    String fuelType;
```
Comment (ssoegriemsingh) [inline 1733Sprint/1-Kenny/vehicles/AstonMartin.java]: Needs a tab?

File: 1733Sprint/1-Kenny/vehicles/Vehicle.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
+package vehicles;
+
+public abstract class Vehicle {
+    protected int speed;
+    protected int seats;
+    protected boolean trunk;
+    protected int wheels;
```
Comment (ssoegriemsingh) [inline 1733Sprint/1-Kenny/vehicles/Vehicle.java]: Are seats, trunk and wheels needed? You never reference them

File: 1733Sprint/1-Kenny/Demo.java (hunk 3 above)
Comment (ohsnapitskenny) [inline 1733Sprint/1-Kenny/Demo.java]: Yes, I can make a move method in the class itself so I don't have to downcast it. Makes the whole thing cleaner. Ty

File: 1733Sprint/1-Kenny/Demo.java (hunk 2 above)
Comment (ohsnapitskenny) [inline 1733Sprint/1-Kenny/Demo.java]: My bad

File: 1733Sprint/1-Kenny/vehicles/Vehicle.java (hunk 5 above)
Comment (ohsnapitskenny) [inline 1733Sprint/1-Kenny/vehicles/Vehicle.java]: It's not needed. I should have removed them

File: 1733Sprint/1-Kenny/vehicles/AstonMartin.java (hunk 4 above)
Comment (ohsnapitskenny) [inline 1733Sprint/1-Kenny/vehicles/AstonMartin.java]: My bad

File: 1733Sprint/1-Kenny/Demo.java (hunk 1 above)
Comment (ohsnapitskenny) [inline 1733Sprint/1-Kenny/Demo.java]: This was the easy way. I should not import packages which I don't use.

File: 1733Sprint/1-Kenny/Demo.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+		    String input = sc.next();
+		    int result = Integer.parseInt(input);
+		    showResult(result);   
+	    } catch (NumberFormatException e) {
+	    	System.out.println("Please enter a number");
+	    }
+    }
+
+    private void showResult(int selection) {
+	    // System.out.println(selection);
```
Comment (ssoegriemsingh) [inline 1733Sprint/1-Kenny/Demo.java]: Remove your "commented out" code

File: 1733Sprint/1-Kenny/vehicles/Ebike.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
+package vehicles;
+
+import vehicles.Bike;
+import vehicles.Electric;
+
+public class Ebike extends Bike implements Electric {
+  	private double batteryLevel;
+    // Constructor
+    public Ebike() {
+    	setBatteryLevel(80);
```
Comment (ssoegriemsingh) [inline 1733Sprint/1-Kenny/vehicles/Ebike.java]: Needs a space :p

File: 1733Sprint/1-Kenny/vehicles/Tesla.java [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+import vehicles.Car;
+import vehicles.Electric;
+
+public class Tesla extends Car implements Electric {
+  	private double batteryLevel;
+    // Constructor
+    public Tesla() {
+    	setBatteryLevel(100);
+    	setSpeed(15);
+        setName("Model X");
```
Comment (ssoegriemsingh) [inline 1733Sprint/1-Kenny/vehicles/Tesla.java]: Needs a backspace?

File: 1733Sprint/1-Kenny/Demo.java [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+			);
+	}
+
+	private void showVehicles() {
+		try {
+			int i = 0;
+			for (Vehicle v : vehicles) {
+				// v = null;
+				System.out.println(i + ") " + vehicles.get(i).getName());
+				// System.out.println("----------------------------------------------------------------");
```
Comment (XDIJK) [inline 1733Sprint/1-Kenny/Demo.java]: Left over comments I assume.

File: 1733Sprint/1-Kenny/Demo.java [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+
+	private void createList() {
+		vehicles.addAll(
+			Arrays.asList(
+				new AstonMartin(),
+				new Tesla(),
+				new Ebike(),
+				new Batavus()
+				)
+			);
```
Comment (XDIJK) [inline 1733Sprint/1-Kenny/Demo.java]: Personally I would've formatted this differently

**Pull request merged**